src/Symfony/Component/HttpKernel/Debug/FileLinkFormatter.php

```diff
@@ -23,10 +23,11 @@ class FileLinkFormatter implements \Serializable
 {
     private $fileLinkFormat;
     private $requestStack;
+    private $router;
     private $baseDir;
-    private $urlFormat;
+    private $queryString;
 
-    public function __construct($fileLinkFormat = null, RequestStack $requestStack = null, $baseDir = null, $urlFormat = null)
+    public function __construct($fileLinkFormat = null, RequestStack $requestStack = null, $baseDir = null, $queryString = null, UrlGeneratorInterface $router = null)
```

Comment:
What about doing what I suggested instead. Keeping the second argument, deprecating the fact that you can pass a RequestStack as a second argument, and accept a UrlGenerator as well? That way, we can remove the possibility to pass a RequestStack in 4.0, which is cleaner.

Reply:
nvm, I've implemented my suggestion in f659954